dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/deduplication/DefaultDeduplicationService.java

```diff
@@ -101,6 +105,70 @@ public void updatePotentialDuplicate( PotentialDuplicate potentialDuplicate )
         potentialDuplicateStore.update( potentialDuplicate );
     }
 
+    @Override
+    public boolean isAutoMergeable( PotentialDuplicate potentialDuplicate )
+    {
+        TrackedEntityInstance trackedEntityInstanceA = Optional.ofNullable( trackedEntityInstanceService
+            .getTrackedEntityInstance( potentialDuplicate.getTeiA() ) )
+            .orElseThrow( () -> new PotentialDuplicateException(
+                "No tracked entity instance found with id '" + potentialDuplicate.getTeiA() + "'." ) );
+
+        TrackedEntityInstance trackedEntityInstanceB = Optional.ofNullable( trackedEntityInstanceService
+            .getTrackedEntityInstance( potentialDuplicate.getTeiB() ) )
+            .orElseThrow( () -> new PotentialDuplicateException(
+                "No tracked entity instance found with id '" + potentialDuplicate.getTeiB() + "'." ) );
+
+        if ( !trackedEntityInstanceA.getTrackedEntityType().equals( trackedEntityInstanceB.getTrackedEntityType() ) )
+        {
+            return false;
+        }
+
+        if ( trackedEntityInstanceA.isDeleted() || trackedEntityInstanceB.isDeleted() )
+        {
+            return false;
+        }
+
+        if ( trackedEntityInstanceA.getRelationshipItems().stream()
+            .anyMatch( relationshipItem -> null != relationshipItem.getRelationship().getTo().getTrackedEntityInstance()
+                && relationshipItem.getRelationship().getTo().getTrackedEntityInstance()
+                    .equals( trackedEntityInstanceB ) ) )
+        {
+            return false;
+        }
+
+        if ( trackedEntityInstanceB.getRelationshipItems().stream()
```

Comment:
Same comment as above.

Also, when you get all relationshipitems from teiA, and see if any leads to teiB. I guess in this case it's about checking both sides (Item 2 from the previous comment). but I think it's better to check both to and from, instead of to from both sides :)